### PR TITLE
[@typescript-eslint/strict-boolean-expressions] First pass of easy fixes

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

The goal is to enable an eslint rule that will error on unnecessary check for always truly/falsy variables.
Some bugs can be avoided by such warnings from the linter, as displayed in this PR (search for `joinTargetWorkspaceAllowed`) or [this](https://github.com/dust-tt/dust/pull/7133) one.

This is a first pass of easy fixes, and there is another one (or two) coming before we can actually enable this rule.

### CODE REVIEW

File: front/.eslintrc.js [hunk 1]
```diff
@@ -18,6 +18,17 @@ module.exports = {
     */
     curly: ["error", "all"],
     "react/no-unescaped-entities": 0,
+    // "@typescript-eslint/strict-boolean-expressions": [
```
Comment (flvndvd) [inline front/.eslintrc.js:21]: ??

File: front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts [hunk 2]
```diff
@@ -58,10 +58,14 @@ async function handler(
 
   switch (req.method) {
     case "GET":
-      const limit = req.query.limit ? parseInt(req.query.limit as string) : 10;
-      const offset = req.query.offset
-        ? parseInt(req.query.offset as string)
-        : 0;
+      const limit =
+        typeof req.query.limit === "string"
+          ? parseInt(req.query.limit as string)
```
Comment (flvndvd) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts:63]:
```suggestion
          ? parseInt(req.query.limit)
```

File: front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts [hunk 3]
```diff
@@ -58,10 +58,14 @@ async function handler(
...
+          : 10;
+      const offset =
+        typeof req.query.offset === "string"
+          ? parseInt(req.query.offset as string)
```
Comment (flvndvd) [inline front/pages/api/poke/workspaces/[wId]/data_sources/[name]/documents/index.ts:67]:
```suggestion
          ? parseInt(req.query.offset)
```

File: front/lib/api/assistant/generation.ts [hunk 4]
```diff
@@ -92,15 +92,16 @@ export async function renderConversationForModelMultiActions({
 
       for (const action of actions) {
         const stepIndex = action.step;
-        steps[stepIndex] = steps[stepIndex] || emptyStep();
+        steps[stepIndex] = stepIndex in steps ? steps[stepIndex] : emptyStep();
```
Comment (flvndvd) [inline front/lib/api/assistant/generation.ts:95]: So this becomes the new way to check if an object has a key. Even though it's probably right, this requires to ensure that all engineers follow this pattern. Did we consider this? https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess

Comment (lasryaric) [reply front/lib/api/assistant/generation.ts:95]: We did consider it yesterday but we have 756 errors to fix if we enable this check, so we could do it but in a separate PR.

Comment (flvndvd) [reply front/lib/api/assistant/generation.ts:95]: But then what's the expectations, should everyone start using this "in" pattern?

Comment (lasryaric) [reply front/lib/api/assistant/generation.ts:95]: Once the linter is on, if you don't use it, you will get an error, so you will have to use it.

**Comment by lasryaric** [comment]
Update
if we enable both the lint rule strict-boolean-expressions and the tsconfig compilerOptions.noUncheckedIndexedAccess, this puts us in an ideal world where always truthy/falsy checks are considered an error, and getting an element from an array or a hasmap returns a T | undefined. We can't easily enable these because then we'll have around 1000 errors to fix in our code base.

**Pull request closed without merging**